saltgui/static/scripts/output.js

```diff
@@ -21,6 +21,63 @@
 
 class Output {
 
+  // format a JSON object
+  // based on an initial indentation and an indentation increment
+  static _formatJSON0(value, indentStr, indentCnt) {
+
+    if(typeof value !== "object") {
+      // a simple type
+      // leave that to the builtin function
+      return JSON.stringify(value);
+    }
+  
+    if(Array.isArray(value)) {
+      // an array
+      // put each element on its own line
+      let str = "[";
+      let separator = "";
+      for(const elem of value) {
+        str += separator + "\n" + indentStr + indentCnt +
+          Output._formatJSON0(elem, indentStr + indentCnt, indentCnt);
+        separator = ",";
+      }
+      if(value.length === 0) {
+        // show the brackets for an empty array a bit wider apart
+        str += " ";
+      } else {
+        str += "\n" + indentStr;
+      }
+      str += "]";
+      return str;
+    }
+  
+    // regular object
+    // put each name+value on its own line
+    const keys = Object.keys(value);
+    let str = "{";
+    let separator = "";
+    for (const [key, val] of Object.entries(value).sort()) {
+      str += separator + "\n" + indentStr + indentCnt + "\"" + key + "\": " +
+        Output._formatJSON0(val, indentStr + indentCnt, indentCnt);
+      separator = ",";
+    }
```

Comment:
Here I see the sorting indeed so that would make sense. But wouldn't it be better if you just sort it and then feed it to `formatJSON` which uses `JSON.stringify` ?

Reply:
(but that was changed due to an other remark)